### PR TITLE
Update to Danish language 10/11-2016

### PR DESCRIPTION
**Pull request by boelle**

### CODE REVIEW

**Review by Kaibob2 — COMMENTED** [review]
@boelle Aren't there a few lines which are too long to be displayed properly on common LCD displays? Only asking.

File: Marlin/language_da.h [hunk 1]
```diff
@@ -178,59 +186,55 @@
...
+#define MSG_INFO_PRINT_COUNT              "Ant. Prints"
+#define MSG_INFO_COMPLETED_PRINTS         "Færdige"
+#define MSG_INFO_PRINT_TIME               "Total print tid"
+#define MSG_INFO_PRINT_LONGEST            "Længste print"
+#define MSG_INFO_PRINT_FILAMENT           "Total Extruderet"
+#define MSG_INFO_PRINT_COUNT              "Prints"
+#define MSG_INFO_COMPLETED_PRINTS         "Færdige"
+#define MSG_INFO_PRINT_TIME               "Total"
+#define MSG_INFO_PRINT_LONGEST            "Længste"
+#define MSG_INFO_PRINT_FILAMENT           "Extruderet"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: These defines are screwed up, and there are two of each. They belong inside the conditionals as you see above. Please put them back as they were.

File: Marlin/language_da.h [hunk 2]
```diff
@@ -45,12 +47,12 @@
 #define MSG_AUTO_HOME_Y                     "Home Y"
 #define MSG_AUTO_HOME_Z                     "Home Z"
 #define MSG_LEVEL_BED_HOMING                "Homing XYZ"
-#define MSG_LEVEL_BED_WAITING               "Klik når du er klar"
+#define MSG_LEVEL_BED_WAITING               "Tryk for at starte bed level"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: These strings are for use on LCD controllers, which have a nominal limit of 20 characters. Translators are asked to strive for 18 or fewer characters, if possible, and 16 or fewer characters ideally.

File: Marlin/language_da.h [hunk 3]
```diff
@@ -45,12 +47,12 @@
...
 #define MSG_LEVEL_BED_NEXT_POINT            "Næste punkt"
 #define MSG_LEVEL_BED_DONE                  "Bed level er færdig!"
 #define MSG_LEVEL_BED_CANCEL                "Annuller bed level"
-#define MSG_SET_HOME_OFFSETS                "Sæt forsk. af home"
-#define MSG_HOME_OFFSETS_APPLIED            "Forsk. er nu aktiv"
+#define MSG_SET_HOME_OFFSETS                "Sæt forskyding af home"
+#define MSG_HOME_OFFSETS_APPLIED            "Forskydninger af home pos. er tilføjet"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: These strings are for use on LCD controllers, which have a nominal limit of 20 characters. Translators are asked to strive for 18 or fewer characters, if possible, and 16 or fewer characters ideally.

File: Marlin/language_da.h [hunk 4]
```diff
@@ -101,6 +102,10 @@
 #define MSG_VZ_JERK                         "Vz-jerk"
 #define MSG_VE_JERK                         "Ve-jerk"
 #define MSG_VMAX                            "Vmax "
+#define MSG_X                               "X"
+#define MSG_Y                               "Y"
+#define MSG_Z                               "Z"
+#define MSG_E                               "E"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: You should remove `MSG_X`, `MSG_Y`, `MSG_Z`, and `MSG_E` (which were intentionally removed).

File: Marlin/language_da.h [hunk 5]
```diff
@@ -140,32 +145,35 @@
...
+#define MSG_BLTOUCH_RESET                   "Reset BLTouch"
 #define MSG_HOME                            "Home"  // Used as MSG_HOME " " MSG_X MSG_Y MSG_Z " " MSG_FIRST
 #define MSG_FIRST                           "først"
 #define MSG_ZPROBE_ZOFFSET                  "Z Offset"
 #define MSG_BABYSTEP_X                      "Babystep X"
 #define MSG_BABYSTEP_Y                      "Babystep Y"
 #define MSG_BABYSTEP_Z                      "Babystep Z"
 #define MSG_ENDSTOP_ABORT                   "Endstop abort"
-#define MSG_HEATING_FAILED_LCD              "Opvarmning fejlet"
+#define MSG_HEATING_FAILED_LCD              "Opvarmning mislykkedes"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: These strings are for use on LCD controllers, which have a nominal limit of 20 characters. Translators are asked to strive for 18 or fewer characters, if possible, and 16 or fewer characters ideally.

File: Marlin/language_da.h [hunk 6]
```diff
@@ -140,32 +145,35 @@
...
-#define MSG_ERR_MAXTEMP                     "Fejl: MAKSTEMP"
-#define MSG_ERR_MINTEMP                     "Fejl: MINTEMP"
-#define MSG_ERR_MAXTEMP_BED                 "Fejl: MAKSTEMP PLADE"
-#define MSG_ERR_MINTEMP_BED                 "Fejl: MINTEMP PLADE"
-#define MSG_HALTED                          "PRINTER HALTED"
-#define MSG_PLEASE_RESET                    "Reset venligst"
-#define MSG_SHORT_DAY                       "d" // One character only
-#define MSG_SHORT_HOUR                      "h" // One character only
-#define MSG_SHORT_MINUTE                    "m" // One character only
+#define MSG_ERR_MAXTEMP                     "Fejl: Maks temp"
```
Comment (thinkyhead) [inline Marlin/language_da.h:163]: This is fine, but I thought we agreed that `MAKSTEMP` and `MINTEMP` were more isomorphic translations.

File: Marlin/language_da.h [hunk 7]
```diff
@@ -140,32 +145,35 @@
...
+#define MSG_ERR_MINTEMP                     "Fejl: Min temp"
+#define MSG_ERR_MAXTEMP_BED                 "Fejl: Maks Plsde temp"
+#define MSG_ERR_MINTEMP_BED                 "Fejl: Min Plade temp"
+#define MSG_ERR_Z_HOMING                    "G28 Z Forbudt"
+#define MSG_HALTED                          "PRINTER STOPPET"
+#define MSG_PLEASE_RESET                    "Reset Venligst"
```
Comment (thinkyhead) [inline Marlin/language_da.h:169]: Check your line-endings and make sure they are "Unix" style (just LF with no CR).

File: Marlin/language_da.h [hunk 8]
```diff
@@ -178,59 +186,55 @@
...
+#define MSG_FILAMENT_CHANGE_EXTRUDE_3       ""
+#define MSG_FILAMENT_CHANGE_RESUME_1        "Vent på at print"
+#define MSG_FILAMENT_CHANGE_RESUME_2        "fortsætter"
+#define MSG_FILAMENT_CHANGE_RESUME_3        ""
+#define MSG_FILAMENT_CHANGE_INIT_1          "Vent venligst..."
+#define MSG_FILAMENT_CHANGE_UNLOAD_1        "Udskyder..."
+#define MSG_FILAMENT_CHANGE_INSERT_1        "Indsæt og klik"
+#define MSG_FILAMENT_CHANGE_LOAD_1          "Indtager..."
+#define MSG_FILAMENT_CHANGE_EXTRUDE_1       "Extrudere..."
+#define MSG_FILAMENT_CHANGE_RESUME_1        "Fortsætter..."
```
Comment (thinkyhead) [inline Marlin/language_da.h]: As in the block above, these belong inside the conditionals. Please restore them as they were.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
@boelle Please make the requested changes.

File: Marlin/language_da.h (hunk 2 above)
Comment (boelle) [reply Marlin/language_da.h]: done

File: Marlin/language_da.h (hunk 3 above)
Comment (boelle) [reply Marlin/language_da.h]: done

File: Marlin/language_da.h (hunk 4 above)
Comment (boelle) [reply Marlin/language_da.h]: done

File: Marlin/language_da.h (hunk 5 above)
Comment (boelle) [reply Marlin/language_da.h]: done

File: Marlin/language_da.h (hunk 6 above)
Comment (boelle) [reply Marlin/language_da.h:163]: i missed that.

File: Marlin/language_da.h (hunk 7 above)
Comment (boelle) [reply Marlin/language_da.h:169]: Blame github. i edit via web exclusive

File: Marlin/language_da.h (hunk 1 above)
Comment (boelle) [reply Marlin/language_da.h]: oki, i just cut out from the english will add the conditionals

File: Marlin/language_da.h (hunk 8 above)
Comment (boelle) [reply Marlin/language_da.h]: will do as above

File: Marlin/language_da.h (hunk 1 above)
Comment (boelle) [reply Marlin/language_da.h]: so like this: 
#ifndef MSG_INFO_MENU
#define MSG_INFO_MENU                       "Om Printer"
#endif

**Comment by boelle** [comment]
about the conditionals... like this:

#ifndef MSG_INFO_MENU
#define MSG_INFO_MENU                       "Om Printer"
#endif

line 183-185

**Comment by boelle** [comment]
@thinkyhead think i got it, might have failed totally

**Comment by boelle** [comment]
@thinkyhead did i get it right on the conditionals ? or are there still faults?

File: Marlin/language_da.h (hunk 1 above)
Comment (thinkyhead) [reply Marlin/language_da.h]: I see you figured out I was referring to

``` cpp
#if LCD_WIDTH > 19
  . . .
#else
  . . .
#endif
```

But you neglected to restore the proper indentation.

File: Marlin/language_da.h (hunk 8 above)
Comment (thinkyhead) [reply Marlin/language_da.h]: And again, formatting counts if you want a passing grade. Please maintain the same indentation.

**Comment by thinkyhead** [comment]
See my additional comments.

File: Marlin/language_da.h (hunk 1 above)
Comment (boelle) [reply Marlin/language_da.h]: i assume you mean i missed 1 space ident

File: Marlin/language_da.h (hunk 8 above)
Comment (boelle) [reply Marlin/language_da.h]: i assume you mean i missed 1 space ident

**Comment by boelle** [comment]
@thinkyhead one space ident added to the 2 conditionals sections

File: Marlin/language_da.h [hunk 9]
```diff
@@ -190,56 +190,56 @@
...
-#define MSG_INFO_PRINT_COUNT              "Ant. Prints"
-#define MSG_INFO_COMPLETED_PRINTS         "Færdige"
-#define MSG_INFO_PRINT_TIME               "Total print tid"
-#define MSG_INFO_PRINT_LONGEST            "Længste print"
-#define MSG_INFO_PRINT_FILAMENT           "Total Extruderet"
+ #define MSG_INFO_PRINT_COUNT              "Ant. Prints"
+ #define MSG_INFO_COMPLETED_PRINTS         "Færdige"
+ #define MSG_INFO_PRINT_TIME               "Total print tid"
+ #define MSG_INFO_PRINT_LONGEST            "Længste print"
+ #define MSG_INFO_PRINT_FILAMENT           "Total Extruderet"
```
Comment (thinkyhead) [inline Marlin/language_da.h]: Please indent with 2 spaces. Thanks! Review the Coding Standards.

File: Marlin/language_da.h [hunk 10]
```diff
@@ -190,56 +190,56 @@
...
+ #define MSG_FILAMENT_CHANGE_LOAD_2          "filament indtag"
+ #define MSG_FILAMENT_CHANGE_LOAD_3          ""
+ #define MSG_FILAMENT_CHANGE_EXTRUDE_1       "Vent på"
+ #define MSG_FILAMENT_CHANGE_EXTRUDE_2       "filament extrudering"
+ #define MSG_FILAMENT_CHANGE_EXTRUDE_3       ""
+ #define MSG_FILAMENT_CHANGE_RESUME_1        "Vent på at print"
+ #define MSG_FILAMENT_CHANGE_RESUME_2        "fortsætter"
+ #define MSG_FILAMENT_CHANGE_RESUME_3        ""
 #endif
 #else // LCD_HEIGHT < 4
```
Comment (thinkyhead) [inline Marlin/language_da.h:236]: Somehow this has both `#endif` and `#else`. Please fix.

File: Marlin/language_da.h [hunk 11]
```diff
@@ -190,56 +190,56 @@
...
-#define MSG_INFO_PRINT_TIME               "Total print tid"
-#define MSG_INFO_PRINT_LONGEST            "Længste print"
-#define MSG_INFO_PRINT_FILAMENT           "Total Extruderet"
+ #define MSG_INFO_PRINT_COUNT              "Ant. Prints"
+ #define MSG_INFO_COMPLETED_PRINTS         "Færdige"
+ #define MSG_INFO_PRINT_TIME               "Total print tid"
+ #define MSG_INFO_PRINT_LONGEST            "Længste print"
+ #define MSG_INFO_PRINT_FILAMENT           "Total Extruderet"
 #endif
 #else
```
Comment (thinkyhead) [inline Marlin/language_da.h:199]: Only `#else` is needed. Somehow an extra `#endif` crept in.

File: Marlin/language_da.h (hunk 9 above)
Comment (thinkyhead) [inline Marlin/language_da.h]: This first group seems to be missing these:

```
MSG_INFO_MIN_TEMP
MSG_INFO_MAX_TEMP
MSG_INFO_PSU
MSG_DRIVE_STRENGTH
MSG_DAC_PERCENT
MSG_DAC_EEPROM_WRITE
MSG_FILAMENT_CHANGE_HEADER
MSG_FILAMENT_CHANGE_OPTION_HEADER
MSG_FILAMENT_CHANGE_OPTION_EXTRUDE
MSG_FILAMENT_CHANGE_OPTION_RESUME
```

Note that for items where the string length doesn't differ, they don't need to be inside the conditional and can be declared just once.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
Please fix the issues I have pointed out. Thanks!

File: Marlin/language_da.h (hunk 9 above)
Comment (boelle) [reply Marlin/language_da.h]: Done

Comment (boelle) [reply Marlin/language_da.h]: I could not find them in the english file which i have copied off

But will add them if i dont hear anything else

File: Marlin/language_da.h (hunk 11 above)
Comment (boelle) [reply Marlin/language_da.h:199]: #endif removed

File: Marlin/language_da.h (hunk 10 above)
Comment (boelle) [reply Marlin/language_da.h:236]: #endif removed

**Comment by boelle** [comment]
@thinkyhead did i pass this time? or is this section in fact missing thou i fail to see it in the english file

```
MSG_INFO_MIN_TEMP
MSG_INFO_MAX_TEMP
MSG_INFO_PSU
MSG_DRIVE_STRENGTH
MSG_DAC_PERCENT
MSG_DAC_EEPROM_WRITE
MSG_FILAMENT_CHANGE_HEADER
MSG_FILAMENT_CHANGE_OPTION_HEADER
MSG_FILAMENT_CHANGE_OPTION_EXTRUDE
MSG_FILAMENT_CHANGE_OPTION_RESUME
```

**Comment by thinkyhead** [comment]
> i fail to see it in the english file

https://github.com/MarlinFirmware/Marlin/blob/RCBugFix/Marlin/language_en.h#L526

**Comment by thinkyhead** [comment]
Rebased, squashed, and patched at #5244

**Pull request closed without merging**